### PR TITLE
fix: remove guava rule

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description

As of guava 32.1, the correct metadata is now published with guava so this is no longer needed. Further, the way we generate our patched metadata is incompatible with those updates leading to failures to resolve guava 32.1. Removing this rather than adjusting the rule to work backwards, since we're upgrading all our guava versions anyway.

### CODE REVIEW

**Review by ravisingal — DISMISSED** [review]

**Comment by aaron-steinfeld** [comment]
@ravisingal @kotharironak looks like will need a review from each due to old codeowners file, updated in this PR.

**Review by ravisingal — APPROVED** [review]

**Review by avinashkolluru — APPROVED** [review]

**Review by kotharironak — APPROVED** [review]

**Pull request merged**
